backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/CpuPinningPolicyUnit.java

```diff
@@ -58,24 +59,33 @@ public CpuPinningPolicyUnit(PolicyUnit policyUnit,
     @Override
     public List<VDS> filter(final SchedulingContext context,
             final List<VDS> hosts,
-            final VM vm,
+            final List<VM> vmGroup,
             final PerHostMessages messages) {
-        final String cpuPinning = vm.getVmPinning();
 
         // return all hosts when no CPU pinning is requested
-        if (vm.getCpuPinningPolicy() == CpuPinningPolicy.NONE) {
+        if (vmGroup.stream().allMatch(vm -> vm.getCpuPinningPolicy() == CpuPinningPolicy.NONE)) {
             return hosts;
         }
 
         // only add hosts as candidates which have all required CPUs up and running
-        final List<VDS> candidates = new ArrayList<>();
+        final List<VDS> candidates = new ArrayList<>(hosts);
```

Comment:
that's a bit unusual - why adding all and then filtering rather than adding the ones that fit?

Reply:
was an easier logic in my eyes :)